packages/hash/frontend/src/lib/entities.ts

```diff
@@ -441,24 +440,48 @@ export const convertApiEntityTypesToBpEntityTypes = (
   records.map((record) => convertApiEntityTypeToBpEntityType(record));
 
 /**
- * This is a temporary solution to guess a display label for an entity.
- * It will be replaced by a 'labelProperty' in the schema indicating which field to use as the label
+ * Generate a display label for an entity
+ * Prefers the BP-specified labelProperty if it exists.
  * @see https://blockprotocol.org/docs/spec/graph-service-specification#json-schema-extensions
  */
-export const guessEntityName = (entity: JsonObject) => {
-  const { name, preferredName, displayName, title, shortname, ...rest } =
-    isParsedJsonObject(entity.properties) ? entity.properties : entity;
-  const valueForKeyContainingName = Object.entries(rest).find(([key]) =>
-    key.toLowerCase().includes("name"),
-  )?.[1];
-  return (
-    name ??
-    preferredName ??
-    displayName ??
-    title ??
-    shortname ??
-    valueForKeyContainingName ??
-    entity.entityId ??
-    "Entity"
-  ).toString();
+export const generateEntityLabel = (
+  entity: Entity,
+  schema?: JsonObject,
```

Comment:
Does it work to do this?

```suggestion
  schema?: { labelProperty?: unknown, title?: unknown },
```

You'll probably get better code intelligence

Reply:
https://github.com/hashintel/hash/pull/1014/commits/0f6b38a572f46b6190083c4edca2231134a1383f